HTTP Basic Auth implementation and passivetotal integration

johannestaas opened the pull request:
Using the --auth/-a flag to pass in an authentication config will allow you to use HTTP basic auth for configs you specify.

Example:

```
$ machinae -a auth.yml passivetotal.org
...
$ cat machinae.yml
...
passivetotal_trackers:
  name: PassiveTotal Trackers
  default: False
  otypes:
    - fqdn
  json:
    request:
      url: 'https://api.passivetotal.org/v2/host-attributes/trackers'
      auth: passivetotal
...
$ cat auth.yml
passivetotal:
  - '<username>'
  - '<api key>'
```

mcm commented:
Wow this is awesome! Unless you want to submit another PR with an update to README.md, I'll get that taken care of and release an updated version with the auth changes.

Thanks so much for the contribution.

johannestaas commented:
Glad you like it!
https://github.com/HurricaneLabs/machinae/pull/19

I added some notes on using authentication and showed an example, and added
PassiveTotal to the list of plugins disabled by default.

Cheers!

On Wed, Apr 20, 2016 at 2:54 PM, mcm <notifications@github.com> wrote:

> Merged #17 <https://github.com/HurricaneLabs/machinae/pull/17>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/HurricaneLabs/machinae/pull/17#event-635244602>
>

azfayel commented:
Hello,
I try to apply the config for using API key with passivetotal and save to auth.yml and it failed to read the file when passing machinae w/ the -a argument.
Can you give more explanation on how to build API key file for having more information from machinae.
